README.md

```diff
@@ -230,6 +230,91 @@ $ puma -C "-"
 
 Check out [dsl.rb](https://github.com/puma/puma/blob/master/lib/puma/dsl.rb) to see all available options.
 
+### TCP server
+
+Puma also could be used as a TCP server to process incoming TCP
+connections.
+
+TCP mode can be enabled with CLI option `--tcp-mode`:
+
+```
+$ puma --tcp-mode
+```
+
+Default ip and port to listen to are `0.0.0.0` and `9292`. You can configure
+them with `--port` and `--bind` options:
+
+```
+$ puma --tcp-mode --bind tcp://127.0.0.1:9293
+$ puma --tcp-mode --port 9293
+```
+
+TCP mode could be set with a configuration file as well with `tcp_mode`
+and `tcp_mode!` methods:
+
+```
+# config/puma.rb
+tcp_mode
+```
+
+When Puma starts in the TCP mode it prints corresponded message:
+
+```
+puma --tcp-mode
+Puma starting in single mode...
+...
+* Mode: Lopez Express (tcp)
+```
+
+Application to process TCP connections should be declared in the same
+way as an ordinary HTTP application. It's a callable object which accepts
```

Comment:
I don't think this is really accurate (that it's created in the same way), because for HTTP Puma serves Rack apps, but for TCP, it's a different structure.